__tests__/overview.test.js

```diff
@@ -1,117 +1,216 @@
 import sampleData from '../client/src/overview/sampleRelevantData.json';
 import carouselData from '../client/src/overview/caroselData.json';
 import sizeData from '../client/src/overview/sizeData.json';
 
-// import { configure, shallow } from 'enzyme';
-// import Adapter from 'enzyme-adapter-react-16';
-// configure({ adapter: new Adapter() });
+import { configure, shallow, mount } from 'enzyme';
+import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
+configure({ adapter: new Adapter() });
 
 afterEach(cleanup);
 
-describe('Overview Component', () => {
-//   describe('Testing Main Component', () => {
-//     test('it renders', () => {
-//       render(<ProductOverview />);
-//       expect(screen.getByTestId('overview')).toBeInTheDocument();
-//     });
-//     test('Loads five sub components', () => {
-//       render(<ProductOverview />);
-
-//       expect(screen.getByTestId('add-to-cart')).toBeInTheDocument();
-//       expect(screen.getByTestId('image-gallery')).toBeInTheDocument();
-//       expect(screen.getByTestId('product-info')).toBeInTheDocument();
-//       expect(screen.getByTestId('style-select')).toBeInTheDocument();
-//       expect(screen.getByTestId('product-description')).toBeInTheDocument();
-//     });
-//   });
-
-  describe('Style Selector renders with Sample Data', () => {
-    test('Renders each style in selector', () => {
-      var styles = sampleData.styles;
-      let styleSelector = render(<StyleSelector styles={styles} currentStyleIndex={0}/>);
-      expect(styleSelector.getAllByRole('listitem').length).toEqual(styles.length);
+describe('Overview Component with Sample Data', () => {
+  describe('Image Gallery Component-normalData', () => {
+    const wrapper = shallow(
+      <ImageGallery
+        photos={sampleData.styles[0].photos}
+        i={0}
+        key={0}
+      />
+    );
+    const thumbnails = wrapper.find('ThumbnailCarousel');
+    const mainImage = wrapper.find('.overview-image-gallery-img');
+    test('Renders Main Image', () => {
+
+      expect(mainImage.length).toBe(1);
     });
-  });
-
-  describe('Image Gallery renders with Sample Data', () => {
-    var photos = sampleData.styles[0].photos;
-    var moreThanSeven = carouselData.styles[0].photos;
-
-    test('Renders all photos', () => {
-      let imageGallery = render(<ImageGallery photos={photos} i={0}/>);
-      expect(imageGallery.getAllByRole('img').length).toEqual(photos.length + 1);
-      expect(imageGallery.getAllByRole('button').length).toEqual(2);
+    test('Renders Thumbnails', () => {
+      expect(thumbnails.length).toBe(1);
     });
-    test('Renders only seven thumbnails at a time', () => {
-      let imageGallery = render(<ImageGallery photos={moreThanSeven} i={0}/>);
-      expect(imageGallery.getAllByRole('img').length).toEqual(8)
+    test('Renders Thumbnail images', () => {
+      const test = thumbnails.getElements();
+      expect(test[0].props.imgs.length).toEqual(sampleData.styles[0].photos.length);
```

Comment:
general comments for image test: checking the length may not be enough, should also check if the src matches, otherwise number of image can be correct but the image content may not be same as expected.

Reply:
You are right, I have updated the test to check for sources!